### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by marcinzaremba**
Part of increasing test coverage initiative.

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: worker.go [hunk 1]
```diff
@@ -149,6 +149,45 @@ func (l *loadBalancer) Owner() string {
 	return ""
 }
 
+// CertificatesFinder interface represents a list of certificates
+// and some basic operations than can be performed on them.
+type CertificatesFinder interface {
```
Comment (mikkeloscar) [inline worker.go]: @marcinzaremba since you refactor this, would it make sense to move it out of `worker.go` to a separate file? `worker.go` is already a bit bloated with stuff :)

Comment (marcinzaremba) [reply worker.go]: I was thinking about this but cannot come up with meaningful separate component (area of responsibility. And I would like not to end up with something pretty meaningless like `worker_utils.go`. Can you maybe suggest some possibilities?

Comment (mikkeloscar) [reply worker.go]: `certificates_finder.go` would be fine for me, it doesn't have to be in a separate package, just to things out of `worker.go`. If you don't want to do it in this PR, it's also totally fine, I just wanted to suggest it :)

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
@mikkeloscar I think it’s fine like this, merge?

**Comment by marcinzaremba** [comment]
Initially I wanted to also test `matchIngressesToLoadBalancers` as I believe this the most critical part of functionality here (with edge cases like out of certificates for LB, shared LB, now security group matching, etc.). I believe this little refactor enables this (provide a mock with interface of `CertificateFinder` to `matchIngressesToLoadBalancers`. So we can merge it in the state as is and let testing this function for others.

**Comment by marcinzaremba** [comment]
👍

**Comment by szuecs** [comment]
Thanks @marcinzaremba great work!
